grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandlerBuilder.java

```diff
@@ -0,0 +1,245 @@
+
+    @Nullable
+    private MessageMarshaller jsonMarshaller;
+
+    @Nullable
+    private List<Message> marshalledMessages;
+
+    @Nullable
+    private List<Class<? extends Message>> marshalledMessageTypes;
+
+    @Nullable
+    private Set<UnframedGrpcErrorResponseType> responseTypes;
+
+    UnframedGrpcErrorHandlerBuilder() {}
+
+    /**
+     * Sets a custom JSON marshaller to be used by the error handler.
+     *
+     * <p>This method allows the caller to specify a custom JSON marshaller
+     * for encoding the error responses. If messages or message types have
+     * already been registered, calling this method will result in an
+     * {@link IllegalStateException}. If nothing is specified,
+     * {@link UnframedGrpcErrorHandlers#ERROR_DETAILS_MARSHALLER} is used as
+     * default json marshaller.
+     *
+     * @param jsonMarshaller The custom JSON marshaller to use
+     */
+    public UnframedGrpcErrorHandlerBuilder jsonMarshaller(MessageMarshaller jsonMarshaller) {
+        requireNonNull(jsonMarshaller, "jsonMarshaller");
+        checkState(marshalledMessages == null && marshalledMessageTypes == null,
+                   "Cannot set a custom JSON marshaller because one or more Message or MessageType instances" +
+                   " have already been registered. To set a custom marshaller, ensure that no Message or" +
+                   " MessageType registrations have been made before calling this method."
+        );
+        this.jsonMarshaller = jsonMarshaller;
+        return this;
+    }
+
+    /**
+     * Specifies the status mapping function to be used by the error handler.
+     *
+     * <p>This function determines how gRPC statuses are mapped to HTTP statuses
+     * in the error response.
+     *
+     * @param statusMappingFunction The status mapping function
+     */
+    public UnframedGrpcErrorHandlerBuilder statusMappingFunction(
+            UnframedGrpcStatusMappingFunction statusMappingFunction) {
+        this.statusMappingFunction = requireNonNull(statusMappingFunction, "statusMappingFunction");
+        return this;
+    }
+
+    /**
+     * Specifies the response types that the error handler will support.
+     *
+     * <p>This method allows specifying one or more response types (e.g., JSON, PLAINTEXT)
+     * that the error handler can produce. If nothing is specified or multiple types are specified, the actual
+     * response type is determined by the response's content type.
+     *
+     * @param responseTypes The response types to support
+     */
+    public UnframedGrpcErrorHandlerBuilder responseTypes(UnframedGrpcErrorResponseType... responseTypes) {
+        requireNonNull(responseTypes, "responseTypes");
+        if (this.responseTypes == null) {
+            this.responseTypes = EnumSet.noneOf(UnframedGrpcErrorResponseType.class);
+        }
+        this.responseTypes.addAll(ImmutableSet.copyOf(responseTypes));
```

Comment:
This could be simplified into:
```suggestion
        Collections.addAll(this.responseTypes, responseTypes);
```